### ISSUE TITLE
Newlines removed in move from doc comment to OpenAPI `description`

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/dropshot#402]
Seems like double newlines are being kept but single newlines are not.

```rs
/// Note that any gaps, unsorted bins, or non-finite values will result in an error.
///
/// Example
/// -------
/// ```rust
/// use oximeter::histogram::{BinRange, Histogram};
```
https://github.com/oxidecomputer/omicron/blob/e9554ad930665e82526b784b1f12300f067298fa/oximeter/oximeter/src/histogram.rs#L201-L222

becomes

```
an error.\n\nExample ------- ```rust use oximeter::histogram::{BinRange, Histogram};\n\nlet edges
```

https://github.com/oxidecomputer/omicron/blob/e9554ad930665e82526b784b1f12300f067298fa/openapi/nexus.json#L8018

**Comment by ahl** [comment]
Tricky! It's hard to distinguish, say, a line-wrapped paragraph from intentional newlines. In this particular case, it seems like the comments are intended for the local crate and aren't appropriate to include in the OpenAPI document at all.

**Comment by bnaecker** [comment]
Yeah, this particular comment predates using a generator, and wasn't originally intended to be part of the OpenAPI spec. I think this one deserves either a wrapper type with a more clear and language-agnostic docstring, or moving most of the existing string to a private comment, e.g., `///` to `//` in the source file.

**Comment by david-crespo** [comment]
Fine with me, seems like most of the fix here is to fix the comment itself.

Maybe a dumb question, but why do we need to process the newlines in the doc comments at all?

https://github.com/oxidecomputer/dropshot/blob/e75844e8ef514521ec60d30d6a1aeb5365b5bab5/dropshot_endpoint/src/lib.rs#L523-L549
